core/src/test/java/org/elasticsearch/search/internal/ShardSearchTransportRequestTests.java

```diff
@@ -96,7 +96,7 @@ private ShardSearchTransportRequest createShardSearchTransportRequest() throws I
             filteringAliases = new AliasFilter(null, Strings.EMPTY_ARRAY);
         }
         return new ShardSearchTransportRequest(searchRequest, shardId,
-                randomIntBetween(1, 100), filteringAliases, Math.abs(randomLong()));
+                randomIntBetween(1, 100), filteringAliases, 1.0f, Math.abs(randomLong()));
```

Comment:
can you randomize the boost here so we also test boost serialization from 6.x to 6.x with a value that's different from default? You could do `randomBoolean() ? 1.0f : randomFloat()` ?